agent/engine/docker_task_engine_test.go

```diff
@@ -1146,3 +1146,27 @@ func TestEngineDisableConcurrentPull(t *testing.T) {
 	assert.False(t, dockerTaskEngine.enableConcurrentPull,
 		"Task engine should not be able to perform concurrent pulling for version < 1.11.1")
 }
+
+// TestTaskWithCircularDependency tests the task with containers of which the
```

Comment:
Weren't this have already tested it: https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/dependencygraph/graph_test.go#L48